Add auto build conf

perlitz opened the pull request:
The pull request adds a method that builds a configuration given the evaluation budget

Commit:
Add run_specs_coarse configuration file

Commit:
Add method to auto build conf file

Commit:
Add method to auto build conf file

Commit:
Add method to auto build conf file

Commit:
Add method to auto build conf file

Commit:
update helm.md

artidoro commented on build_run_specs_full.py:
```diff
@@ -0,0 +1,237 @@
+    total_n_examples = 500
+    max_eval_instances_per_scenario = total_n_examples//len(df.scenario.unique())
+
+    for scenario, n_sucscenarios in scenarios.items():
```
Just a typo here.

perlitz replied:
will fix

artidoro commented on build_run_specs_full.py:
```diff
@@ -0,0 +1,237 @@
+    scenarios = df.value_counts('scenario').to_dict()
+
+    total_n_examples = 500
+    max_eval_instances_per_scenario = total_n_examples//len(df.scenario.unique())
```
You will have fewer than `total_n_examples` if the division has a remainder, which is the case for both 500 and 600. 
For example, when running with `total_n_examples` of 600, you will end up with 570 total examples. Maybe it's worth distributing remainder somehow like you were doing earlier, or change the name of the file to the actual number of examples that will be run.

perlitz replied:
good idea, I will distribute these

artidoro reviewed (approved):
Some small comments but overall I think this will be useful to speed up eval time.

weiweiy commented on build_run_specs_full.py:
```diff
@@ -0,0 +1,237 @@
+    {'scenario':'operator','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=operators,subtask=", 'priority': 1},
+
+    # 17. parsinlu_reading_comprehension: https://github.com/google/big-bench/tree/main/bigbench/benchmark_tasks/parsinlu_reading_comprehension
+    {'scenario':'parsinlu_reading_comprehension','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=parsinlu_reading_comprehension,subtask=", 'priority': 1},
```
I propose remove this Persian language task

perlitz replied:
will do

weiweiy commented on build_run_specs_full.py:
```diff
@@ -0,0 +1,237 @@
+    {'scenario':'logical_deduction','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=logical_deduction,subtask=seven_objects", 'priority': 1},
+
+    # 14. misconceptions_russian: https://github.com/google/big-bench/tree/main/bigbench/benchmark_tasks/misconceptions_russian
+    {'scenario':'misconceptions_russian','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=misconceptions_russian,subtask=", 'priority': 1},
```
Remove Russian Specific task

perlitz replied:
will do

weiweiy commented on build_run_specs_full.py:
```diff
@@ -0,0 +1,237 @@
+    {'scenario':'formal_fallacies_syllogisms_negation','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=formal_fallacies_syllogisms_negation,subtask=", 'priority': 1},
+
+    # 8. hindu_knowledge: https://github.com/google/big-bench/tree/main/bigbench/benchmark_tasks/hindu_knowledge
+    {'scenario':'hindu_knowledge','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=hindu_knowledge,subtask=", 'priority': 1},
```
Remove Hindu specific task

perlitz replied:
will do

weiweiy commented on build_run_specs_full.py:
```diff
@@ -0,0 +1,237 @@
+    {'scenario':'conceptual_combinations','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=conceptual_combinations,subtask=homonyms", 'priority': 1},
+    {'scenario':'conceptual_combinations','description': "big_bench:model=neurips/local,max_train_instances=big_bench_few_shot_setting,task=conceptual_combinations,subtask=invented_words", 'priority': 1},
+
+    # 5. conlang_translation: https://github.com/google/big-bench/tree/main/bigbench/benchmark_tasks/conlang_translation
```
@artidoro what your feelings on these Conlang tasks?

weiweiy commented on build_run_specs_full.py:
```diff
@@ -0,0 +1,237 @@
+entries = [
+    #bigbench
+    # 1. auto_debugging: https://github.com/google/big-bench/tree/main/bigbench/benchmark_tasks/auto_debugging
```
@msaroufim  @artidoro, how you feel about code understanding type of tasks for auto_debugging and code description task?

weiweiy commented:
@perlitz , Thank you for creating this PR!
In general, I propose to remove Hindu, Russian and Persian specific tasks... however I am somewhat tore when it comes to code understanding (auto_debugging / code_line_description) and Colang tasks.  
@artidoro @jisaacso @msaroufim  thoughts?

msaroufim commented:
personally i think the coding and conlang tasks are interesting

Commit:
Improve conf generation script
- Allow to run from terminal
- Add documentation
- Use all examples in the budget (https://github.com/llm-efficiency-challenge/neurips_llm_efficiency_challenge/pull/35#discussion_r1337759954)
- Remove some scenarios

weiweiy reviewed (approved):
@perlitz  Thanks again for making this PR and addressing our issues.
